src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/WSSecurityXXX2005.cs

```diff
@@ -0,0 +1,249 @@
+            }
+
+            public override void WriteKeyIdentifierClauseCore(XmlDictionaryWriter writer, SecurityKeyIdentifierClause keyIdentifierClause)
+            {
+                for (int i = 0; i < this.StrEntries.Count; ++i)
+                {
+                    if (this.StrEntries[i].SupportsCore(keyIdentifierClause))
+                    {
+                        writer.WriteStartElement(CoreWCF.XD.SecurityJan2004Dictionary.Prefix.Value, CoreWCF.XD.SecurityJan2004Dictionary.SecurityTokenReference, CoreWCF.XD.SecurityJan2004Dictionary.Namespace);
+
+                        string tokenTypeUri = this.GetTokenTypeUri(this.StrEntries[i], keyIdentifierClause);
+                        if (tokenTypeUri != null)
+                        {
+                            writer.WriteAttributeString(CoreWCF.XD.SecurityXXX2005Dictionary.Prefix.Value, CoreWCF.XD.SecurityXXX2005Dictionary.TokenTypeAttribute, CoreWCF.XD.SecurityXXX2005Dictionary.Namespace, tokenTypeUri);
+                        }
+
+                        this.StrEntries[i].WriteContent(writer, keyIdentifierClause);
+                        writer.WriteEndElement();
+                        return;
+                    }
+                }
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.StandardsManagerCannotWriteObject, keyIdentifierClause.GetType())));
+            }
+
+            private string GetTokenTypeUri(StrEntry str, SecurityKeyIdentifierClause keyIdentifierClause)
+            {
+                bool emitTokenType = this.EmitTokenType(str);
+                if (emitTokenType)
+                {
+                    string tokenTypeUri;
+                    if (str is LocalReferenceStrEntry)
+                    {
+                        tokenTypeUri = (str as LocalReferenceStrEntry).GetLocalTokenTypeUri(keyIdentifierClause);
+                        // only emit token type for SAML,Kerberos and Encrypted References
+                        switch (tokenTypeUri)
+                        {
+                            case SecurityXXX2005Strings.Saml20TokenType:
+                            case SecurityXXX2005Strings.SamlTokenType:
+                            case SecurityXXX2005Strings.EncryptedKeyTokenType:
+                            case SecurityJan2004Strings.KerberosTokenTypeGSS: break;
+
+                            default:
+                                tokenTypeUri = null;
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        tokenTypeUri = str.GetTokenTypeUri();
+                    }
+
+                    return tokenTypeUri;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            private bool EmitTokenType(StrEntry str)
+            {
```

Comment:
In the WCF client I replaced this method body with a comment saying the only token types which need to have the type emitted aren't supported on .NET Core and I simply return false.